File: P004/main.py
import time

start = time.time()

#def is_palindrome2(num):
#    if str(num)==str(num)[::-1]: return True
#            else: return False
#Faster way found on code.jasonbhill.com

def is_palindrome(n):
    oldNum = n
    newNum = 0

    while n >= 1:
        newNum = (newNum*10) + n%10
        n /= 10

    if oldNum == newNum: return True
    else: return False

a = 0
for i in range(100, 999):
    for j in range(i, 999):
        calc = i*j
        if calc > a and is_palindrome(calc):
            a = calc

print a

elapsed = time.time() - start

print("found in %s seconds") % (elapsed)

File: P004/main.java
import java.lang.*;

public class main {

    private static boolean is_palindrome(int n) {
        int oldNum = n;
        int newNum = 0;

        while(n >= 1) {
            newNum = (newNum*10) + n%10;
            n/=10;
        }

        if(oldNum == newNum)
            return true;
        else
            return false;
    }

    public static void main(String[] args) {
        long startTime = System.currentTimeMillis();

        int a = 0;
        for(int i = 100; i<1000; i++) {
            for(int j = i; j<1000; j++) {
                int calc = i*j;
                if(calc > a && is_palindrome(calc)) {
                    a = calc;
                }
            }
        }

        System.out.println(a);


        long endTime = System.currentTimeMillis();
        long totalSpan = System.currentTimeMillis();
        System.out.println("Completed in " + totalSpan + " milliseconds.");
    }
}
